From 8ea3b85c51ae46ba2d97f5e97ef13454fdf98d14 Mon Sep 17 00:00:00 2001
From: Dave Eddy <dave@daveeddy.com>
Date: Fri, 23 Feb 2018 12:17:52 -0500
Subject: [PATCH] TRITON-172 vmapi vms.list.test.js failure when validating
 brands Reviewed by: Josh Wilsdon <josh@wilsdon.ca> Approved by: Josh Wilsdon
 <josh@wilsdon.ca>

---
 lib/common/validation.js | 4 +++-
 test/vms.list.test.js    | 4 ++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/lib/common/validation.js b/lib/common/validation.js
index d990c16..0a42d1c 100644
--- a/lib/common/validation.js
+++ b/lib/common/validation.js
@@ -620,6 +620,8 @@ function createValidateStringsListFn(field, list, options) {
     assert.string(field, 'field');
     assert.arrayOfString(list, 'list');
 
+    var sortedList = jsprim.deepCopy(list).sort();
+
     return function (params) {
         var errs = [];
 
@@ -627,7 +629,7 @@ function createValidateStringsListFn(field, list, options) {
             errs.push(errors.missingParamErr(field));
 
         if (params[field] !== undefined && list.indexOf(params[field]) === -1) {
-            var message = 'Must be one of: ' + list.join(', ');
+            var message = 'Must be one of: ' + sortedList.join(', ');
             errs.push(errors.invalidParamErr(field, message));
         }
 
diff --git a/test/vms.list.test.js b/test/vms.list.test.js
index 164041e..5c45094 100644
--- a/test/vms.list.test.js
+++ b/test/vms.list.test.js
@@ -117,7 +117,7 @@ var VALID_VM_BRANDS = [
     'joyent-minimal',
     'joyent',
     'kvm'
-];
+].sort();
 
 exports.list_param_valid_brands = function (t) {
     async.each(VALID_VM_BRANDS, function (vmBrand, next) {
@@ -218,7 +218,7 @@ var VALID_VM_STATES = [
     'stopped',
     'active',
     'destroyed'
-];
+].sort();
 
 exports.list_param_valid_state = function (t) {
     async.each(VALID_VM_STATES, function (vmState, next) {
-- 
2.21.0

